File: android/smartfoo-android-lib-core/src/main/java/com/smartfoo/android/core/FooAssert.java
package com.smartfoo.android.core;

public class FooAssert {
    public static void assertThrows(Runnable runnable, Class<IllegalArgumentException> illegalArgumentExceptionClass) {
        try {
            runnable.run();
            throw new AssertionError("Expected exception not thrown");
        } catch (Exception e) {
            if (illegalArgumentExceptionClass.isInstance(e)) {
                return;
            }
            throw e;
        }
    }
}

File: android/smartfoo-android-lib-core/src/main/java/com/smartfoo/android/core/FooAssert.kt
package com.smartfoo.android.core

/**
 * Lightweight assertion helpers for use in tests and debug validation.
 */
object FooAssert {
    /**
     * Asserts that executing [runnable] throws an exception that is an instance of
     * [illegalArgumentExceptionClass]. Throws [AssertionError] if no exception is thrown, or
     * rethrows the exception if it is not of the expected type.
     *
     * @param runnable the code block expected to throw
     * @param illegalArgumentExceptionClass the expected exception class
     * @throws AssertionError if [runnable] completes without throwing
     */
    fun assertThrows(
        runnable: Runnable,
        illegalArgumentExceptionClass: Class<IllegalArgumentException?>
    ) {
        try {
            runnable.run()
            throw AssertionError("Expected exception not thrown")
        } catch (e: Exception) {
            if (illegalArgumentExceptionClass.isInstance(e)) {
                return
            }
            throw e
        }
    }
}
